From af9901f7ac03fa09ec4268cec1cea8fbdf40c58c Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Mon, 14 Jul 2025 15:12:53 +0200
References: git-fixes
Patch-mainline: Never (kABI fixup)
Subject: [PATCH] ucsi_operations: add stubs for all operations

Having used up a padding to allow more operations, it
makes sense to add stubs for all methods currently
upstream and new placeholders.

Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/usb/typec/ucsi/ucsi.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/usb/typec/ucsi/ucsi.h b/drivers/usb/typec/ucsi/ucsi.h
index 122adfc8f..50c90c542 100644
--- a/drivers/usb/typec/ucsi/ucsi.h
+++ b/drivers/usb/typec/ucsi/ucsi.h
@@ -72,6 +72,19 @@ struct dentry;
 struct ucsi_operations_suse {
 	void (*connector_status)(struct ucsi_connector *con);
 	void (*update_connector)(struct ucsi_connector *con);
+	int (*read_version)(struct ucsi *ucsi, u16 *version);
+	int (*read_cci)(struct ucsi *ucsi, u32 *cci);
+	int (*poll_cci)(struct ucsi *ucsi, u32 *cci);
+	int (*read_message_in)(struct ucsi *ucsi, void *val, size_t val_len);
+	int (*sync_control)(struct ucsi *ucsi, u64 command, u32 *cci,
+			    void *data, size_t size);
+	int (*async_control)(struct ucsi *ucsi, u64 command);
+
+	void *suse_kabi_padding1;
+	void *suse_kabi_padding2;
+	void *suse_kabi_padding3;
+	void *suse_kabi_padding4;
+	void *suse_kabi_padding5;
 };
 
 struct ucsi_operations {
-- 
2.50.1

